ebib-utils.el

```diff
@@ -392,22 +392,38 @@ transformation function returns something that can be displayed."
   :type '(repeat (cons (string :tag "Field")
                        (function :tag "Transform function"))))
 
-(defcustom ebib-TeX-markup-replace-alist '(("{" . "")
-                                           ("}" . "")
-                                           ("``" . "“")
+(defcustom ebib-TeX-markup-replace-alist '(("``" . "“")
 					   ("`" . "‘")
 					   ("''" . "”")
-					   ("'" . "’"))
+					   ("'" . "’")
+					   ("\\\\textit\\(?:\\[.*\\]\\)?{\\([^}]*\\)}"
+					    . (lambda (str)
+						(propertize
+						 (match-string 1 str) 'face 'italic)))
+					    ("\\\\emph\\(?:\\[.*\\]\\)?{\\([^}]*\\)}"
+					    . (lambda (str)
+						(propertize
+						 (match-string 1 str) 'face 'italic)))
+					   ("\\\\textbf\\(?:\\[.*\\]\\)?{\\([^}]*\\)}"
+					    . (lambda (str)
+						(propertize
+						 (match-string 1 str) 'face 'bold)))
+					   ("\\\\textsc\\(?:\\[.*\\]\\)?{\\([^}]*\\)}"
+					    . (lambda (str) (upcase (match-string 1 str))))
+					   ("{" . "")
+                                           ("}" . ""))
   "Alist of strings and replacements for TeX markup.
 This is used in `ebib-clean-TeX-markup' to make TeX markup more
-suitable for display.  Each item in the list consists of a string
-and its replacement.  Both must be strings.  Earlier elements are
-evaluated before later ones, so if one string is a subpattern of
-another, the second must appear later (e.g. \"''\" is before
-\"'\"."
+suitable for display.  Each item in the list consists of a regexp
+and its replacement.  The replacement can be a string (which will
+simply replace the match) or a function (the match will be
+replaced by the result of calling the function on the match
+string).  Earlier elements are evaluated before later ones, so if
+one string is a subpattern of another, the second must appear
+later (e.g. \"''\" is before \"'\")."
   :group 'ebib
-  :type '(alist (string :tag "String")
-		(string :tag "Replacement")))
+  :type '(alist (regexp :tag "Pattern")
+		((choice string function) :tag "Replacement")))
```

Comment:
It might be nice to use a somewhat more user-friendly display in customization buffers, e.g. a set of check boxes to turn different options on or off. But that can be a later improvement, doesn't have to be in this PR.

Reply:
Well, I do this in `writeroom-mode` here: https://github.com/joostkremers/writeroom-mode/blob/eac1da790f316f357ed76ed67fbb790d6a4d126a/writeroom-mode.el#L256, which I think was inspired by some Org mode customization options.

I'm not entirely sure it can be adapted to work in this case and what it would look like in the Customize buffer, but it may be worth looking into at some point. (Feel free to leave it to me, though. Customize is quite a beast, I'd understand if you wouldn't want to invest your time in it. :smile:)

Reply:
It then looks like this:

![image](https://user-images.githubusercontent.com/191630/158810783-9942c5a5-4b9f-4645-af53-69ed17bf1eb6.png)